Source/Core/VideoBackends/Software/SWVertexLoader.cpp

```diff
@@ -230,6 +228,18 @@ void SWVertexLoader::ParseVertex(const PortableVertexDeclaration& vdec, int inde
   {
     ReadVertexAttribute<float>(&m_vertex.normal[i][0], src, vdec.normals[i], 0, 3, false);
   }
+  if (!vdec.normals[1].enable)
+  {
+    m_vertex.normal[1][0] = VertexShaderManager::constants.cached_binormal[0];
+    m_vertex.normal[1][1] = VertexShaderManager::constants.cached_binormal[1];
+    m_vertex.normal[1][2] = VertexShaderManager::constants.cached_binormal[2];
+  }
+  if (!vdec.normals[2].enable)
+  {
+    m_vertex.normal[2][0] = VertexShaderManager::constants.cached_tangent[0];
+    m_vertex.normal[2][1] = VertexShaderManager::constants.cached_tangent[1];
+    m_vertex.normal[2][2] = VertexShaderManager::constants.cached_tangent[2];
+  }
```

Comment:
I've just realized that these names are backwards; based on the order the components are listed in table III of [US6424348B2](https://patents.google.com/patent/US6424348B2/en), it's normal, tangent, binormal (although other parts refer to `GX_VA_NBT` and normal/binormal/tangent).  See also [this code](https://github.com/dolphin-emu/dolphin/blob/3ed9d5a3c7b2426cfbec5ae97df3e5fa6b5472d1/Source/Core/VideoBackends/Software/TransformUnit.cpp#L120-L128) and the [values for `BinormalT` and `BinormalB`](https://github.com/dolphin-emu/dolphin/blob/3ed9d5a3c7b2426cfbec5ae97df3e5fa6b5472d1/Source/Core/VideoCommon/XFMemory.h#L80-L83).

Since I've consistently had binormal be at index 1 and tangent at index 2, this doesn't result in any emulation difference, but I should probably swap the names for consistency.

Reply:
I've swapped the names.